in_toto/runlib.go

```diff
@@ -1,13 +1,20 @@
 package in_toto
 
 import (
+	"errors"
 	"io/ioutil"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"syscall"
 )
 
+// ErrSymCycle signals a detected symlink cycle in our RecordArtifacts() function.
+var ErrSymCycle = errors.New("symlink cycle detected")
+
+// visitedSymlinks is a hashset that contains all paths that we have visited.
+var visitedSymlinks map[string]struct{}
```

Comment:
FYI, we have a nice custom `Set` implementation with `Add` and `Has` methods in `util.go`. :) I suggest you use that.

Reply:
Ah cool, yes this makes sense. Would increase the readability a lot.

Reply:
I've implemented a wrapper around recordArtifacts + use the Set from utils.